docs/WATSONX_EVALUATION_README.md

```diff
@@ -0,0 +1,243 @@
+
+We have successfully implemented **automatic evaluation middleware** that seamlessly integrates with existing Maestro agents without requiring any code changes. **The system is fully functional and returning actual metric scores!**
+
+### ✅ What's Working
+
+#### 1. **Automatic Evaluation Middleware**
+- **Zero Configuration**: Just set `MAESTRO_AUTO_EVALUATION=true` to enable
+- **Transparent Integration**: Works with any existing agent (`MockAgent`, `OpenAIAgent`, etc.)
+- **Production Ready**: Graceful fallback when watsonx library isn't available
+- **Clean Implementation**: Minimal debug output, production-ready code
+- **Real Metrics**: Returns actual numerical scores from watsonx evaluation
+
+#### 2. **Watsonx Integration**
+- **Decorator Pattern**: Successfully implemented IBM watsonx's decorator-based evaluation
+- **Authentication**: Integrated with IBM Cloud API authentication
+- **Multiple Metrics**: Evaluates:
+  - **Answer Relevance**: Measures how well the response addresses the question
+  - **Faithfulness**: Measures how faithful the response is to provided context
+  - **Context Relevance**: Measures how relevant the context is to the question
+  - **Answer Similarity**: Measures similarity between expected and actual answers
+- **Actual Scores**: Returns real numerical scores (0.0-1.0) with evaluation methods
+
+#### 2.1. **Available Metrics (Current)**
+
+| Metric              | Needs Context? | Needs Expected Answer? | Notes |
+|---------------------|----------------|-------------------------|-------|
+| Answer Relevance    | No             | No                      | Response addresses the question |
+| Faithfulness        | Yes            | No                      | Response grounded in provided context |
+| Context Relevance   | Yes            | No                      | Context relevance to the question |
+| Answer Similarity   | No             | Yes                     | Similarity to a reference answer |
+
+Inputs:
+- Context must be a list of strings (e.g., `["doc chunk 1", "doc chunk 2"]`).
+- Expected answer is a single string.
+- If a required input is missing, that metric is skipped gracefully.
+
+#### 3. **Environment Setup**
+- **Python 3.11**: Dedicated `.venv-eval` environment for watsonx compatibility
+- **Dual Environment**: Separate evaluation environment from main development
+- **Error Handling**: Robust error handling for missing dependencies
+- **Environment Variables**: Proper integration with `WATSONX_APIKEY` requirement
+
+### 🔧 Current Implementation
+
+#### Core Files
+```
+src/maestro/agents/evaluation_middleware.py   # Main evaluation middleware
+src/maestro/agents/mock_agent.py             # Updated with middleware integration
+tests/yamls/agents/evaluation_test_agent.yaml      # Test agent configuration
+tests/yamls/workflows/evaluation_test_workflow.yaml # Test workflow
+```
+
+#### How It Works
```

Comment:
can we expand on this with links. Without context about watsonx it's not clear what the decorators or middleware are